NATIVE-167: Add documentation for design-changes and persistence (#36)
Integreat/integreat-react-native-app: docs/11-persistence.md

# Persistence

The structure of the data stored on the device is:

```
    ├── cities.json
    ├── content
    │   └── augsburg
    │       └── en
    │           ├── categories.json
    │           └── events.json
    └── resource-cache
        └── augsburg
            ├── files
            │   └── test.jpg
            └── files.json
```

The `cities.json` could contain information about all cities. Usually they are not persisted:
```
{
  "regensburg": {
    "name": "Stadt Regensburg",
    "languages": [
      {
        "code": "de",
        "name": "Deutsch",
        "dir": "ltr"
      }
    ],
    "offline_available": true,
    "live": true,
    "sorting_name": "Regensburg",
    "events": true,
    "push_notifications": true,
    "extras": true
  }
}
```

The ''content'' is responsible to store the JSON data from the API.

> The ''resource-cache'' contains all the resources. Resources are files which are referenced in the HTML code and thumbnail of pages.

Reviewer: ```suggestion
The ''resource-cache'' contains all the resources. Resources are files which are either referenced in the HTML content or used as thumbnails of pages.
```
